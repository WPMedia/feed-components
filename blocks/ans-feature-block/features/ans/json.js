import Consumer from 'fusion:consumer'
import getProperties from 'fusion:properties'
import { resizerKey } from 'fusion:environment'
import { buildResizerURL } from '@wpmedia/feeds-resizer'

const jmespath = require('jmespath')
export function ANSFeed({ globalContent, customFields, arcSite }) {
  const { resizerURL = '', feedDomainURL = '' } = getProperties(arcSite)
  const { width = 0, height = 0 } = customFields.resizerKVP || {}

  const resizeImage = (img) => {
    if (img && img.url) {
      if (
        img.additional_properties &&
        img.additional_properties.fullSizeResizeUrl
      ) {
        img.url = `${feedDomainURL}${img.additional_properties.fullSizeResizeUrl}`
      } else {
        img.url = buildResizerURL(
          img.url,
          resizerKey,
          resizerURL,
          width,
          height,
        )
      }
      return img
    }
  }

  const resizedContent = (
    jmespath.search(globalContent, 'content_elements||children') || []
  ).map((i) => {
    i.promo_items &&
      Object.keys(i.promo_items).forEach((e) => {
        const promo = i.promo_items[e]
        const resizedPromo = resizeImage(promo)
        if (resizedPromo) i.promo_items[e] = resizedPromo
      })
    i.content_elements &&
      i.content_elements.map((e) => {
        switch (e.type) {
          case 'image':
            e = resizeImage(e)
            break
          case 'gallery':
            e.content_elements.forEach((i) => resizeImage(i))
            break
        }
        return e
      })
    i.related_items &&
      Object.keys(i.related_content).forEach((k) => {
        i.related_content[k].map((e) => {
          switch (e.type) {
            case 'image':
              e = resizeImage(e)
              break
            case 'gallery':
              e.content_elements.forEach((i) => resizeImage(i))
              break
          }
          return e
        })
      })
    i.promo_image &&
      i.promo_image.url &&
      (i.promo_image = resizeImage(i.promo_image))
    return i
  })
  return [resizedContent || []]
}

ANSFeed.label = 'ANS'
export default Consumer(ANSFeed)
