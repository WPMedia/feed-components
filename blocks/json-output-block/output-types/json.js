const Json = ({ children }) => {
  return Array.isArray(children) ? children[0] : null
}

Json.contentType = 'application/json'
export default Json
