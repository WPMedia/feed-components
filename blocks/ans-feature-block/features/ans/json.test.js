// eslint-disable-next-line no-unused-vars
import Consumer from 'fusion:consumer'
import { ANSFeed } from './json'

it('returns template with no values', () => {
  const ans = ANSFeed({
    arcSite: 'the-globe',
    globalContent: {},
    customFields: {},
  })
  expect(ans).toMatchSnapshot()
})

it('returns template with default values', () => {
  const ans = ANSFeed({
    arcSite: 'the-globe',
    globalContent: {
      content_elements: [
        {
          type: 'story',
          last_updated_date: '2020-04-07T15:02:08.918Z',
          website_url: '/food/2020/04/07/tips-for-safe-hand-washing',
          promo_items: {
            basic: {
              title: 'Hand Washing',
              url: 'https://arc-anglerfish-arc2-prod-demo.s3.amazonaws.com/public/JTWX7EUOLJE4FCHYGN2COQAERY.png',
            },
          },
        },
        {
          type: 'story',
          last_updated_date: '2021-04-07T17:02:08.918Z',
          website_url: '/food/2021/04/07/will-we-ever-stop-hand-washing',
          content_elements: [
            {
              type: 'image',
              url: 'https://arc-anglerfish-arc2-prod-demo.s3.amazonaws.com/public/JTWX7EUOLJE4FCHYGN2COQAERY.png',
            },
          ],
        },
        {
          type: 'video',
          last_updated_date: '2021-04-03T13:02:08.918Z',
          website_url: '/food/2021/04/03/best-sourdough-recipes',
          promo_items: { basic: { type: 'video' } },
          promo_image: {
            title: 'No kneed recipes',
            url: 'https://arc-anglerfish-arc2-prod-demo.s3.amazonaws.com/public/JTWX7EUOLJE4FCHYGN2COQAERY.png',
          },
        },
      ],
    },
    customFields: {},
  })
  expect(ans).toMatchSnapshot()
})

it('returns template with site service values', () => {
  const ans = ANSFeed({
    arcSite: 'the-globe',
    globalContent: {
      _id: '/',
      children: [
        { _id: '/channels', children: [{ _id: '/channels/subchannels' }] },
      ],
    },
    customFields: {},
  })
  expect(ans).toMatchSnapshot()
})
