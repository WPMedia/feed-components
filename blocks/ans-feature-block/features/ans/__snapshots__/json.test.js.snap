// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ANS undefined test 1`] = `
Array [
  Array [
    Object {},
  ],
]
`;

exports[`returns ANS for story 1`] = `
Array [
  Array [
    Object {
      "_id": "ABCD1234",
      "content_elements": Array [
        Object {
          "content": "body goes here",
          "type": "text",
        },
        Object {
          "content": "second paragraph goes here",
          "type": "text",
        },
      ],
      "last_updated_date": "2020-04-07T15:02:08.918Z",
      "promo_items": Object {
        "basic": Object {
          "title": "Hand Washing",
          "url": "hi/abcdefghijklmnopqrstuvwxyz=/arc-anglerfish-arc2-prod-demo.s3.amazonaws.com/public/JTWX7EUOLJE4FCHYGN2COQAERY.png",
        },
      },
      "type": "story",
      "website_url": "/food/2020/04/07/tips-for-safe-hand-washing",
    },
  ],
]
`;

exports[`returns ANS from navigation set 1`] = `
Array [
  Array [
    Object {
      "_id": "/channels",
      "_website": "demo",
      "children": Array [
        Object {
          "_id": "/channels/ms-teams",
          "_website": "demo",
          "children": Array [],
          "inactive": false,
          "name": "MS Teams",
          "node_type": "section",
        },
        Object {
          "_id": "/channels/slack",
          "_website": "demo",
          "children": Array [],
          "inactive": false,
          "name": "Slack",
          "node_type": "section",
        },
        Object {
          "_id": "/channels/twitter",
          "_website": "demo",
          "children": Array [],
          "inactive": false,
          "name": "Twitter",
          "node_type": "section",
        },
      ],
      "inactive": false,
      "name": "Channels",
      "node_type": "section",
    },
  ],
]
`;

exports[`returns ANS from results set 1`] = `
Array [
  Array [
    Object {
      "last_updated_date": "2020-04-07T15:02:08.918Z",
      "promo_items": Object {
        "basic": Object {
          "title": "Hand Washing",
          "url": "hi/abcdefghijklmnopqrstuvwxyz=/arc-anglerfish-arc2-prod-demo.s3.amazonaws.com/public/JTWX7EUOLJE4FCHYGN2COQAERY.png",
        },
      },
      "type": "story",
      "website_url": "/food/2020/04/07/tips-for-safe-hand-washing",
    },
    Object {
      "content_elements": Array [
        Object {
          "type": "image",
          "url": "hi/abcdefghijklmnopqrstuvwxyz=/arc-anglerfish-arc2-prod-demo.s3.amazonaws.com/public/JTWX7EUOLJE4FCHYGN2COQAERY.png",
        },
      ],
      "last_updated_date": "2021-04-07T17:02:08.918Z",
      "type": "story",
      "website_url": "/food/2021/04/07/will-we-ever-stop-hand-washing",
    },
    Object {
      "last_updated_date": "2021-04-03T13:02:08.918Z",
      "promo_image": Object {
        "title": "No kneed recipes",
        "url": "hi/abcdefghijklmnopqrstuvwxyz=/arc-anglerfish-arc2-prod-demo.s3.amazonaws.com/public/JTWX7EUOLJE4FCHYGN2COQAERY.png",
      },
      "promo_items": Object {
        "basic": Object {
          "type": "video",
        },
      },
      "type": "video",
      "website_url": "/food/2021/04/03/best-sourdough-recipes",
    },
  ],
]
`;

exports[`returns template with default values 1`] = `
Array [
  Array [
    Object {
      "last_updated_date": "2020-04-07T15:02:08.918Z",
      "promo_items": Object {
        "basic": Object {
          "title": "Hand Washing",
          "url": "hi/abcdefghijklmnopqrstuvwxyz=/arc-anglerfish-arc2-prod-demo.s3.amazonaws.com/public/JTWX7EUOLJE4FCHYGN2COQAERY.png",
        },
      },
      "type": "story",
      "website_url": "/food/2020/04/07/tips-for-safe-hand-washing",
    },
    Object {
      "content_elements": Array [
        Object {
          "type": "image",
          "url": "hi/abcdefghijklmnopqrstuvwxyz=/arc-anglerfish-arc2-prod-demo.s3.amazonaws.com/public/JTWX7EUOLJE4FCHYGN2COQAERY.png",
        },
      ],
      "last_updated_date": "2021-04-07T17:02:08.918Z",
      "type": "story",
      "website_url": "/food/2021/04/07/will-we-ever-stop-hand-washing",
    },
    Object {
      "last_updated_date": "2021-04-03T13:02:08.918Z",
      "promo_image": Object {
        "title": "No kneed recipes",
        "url": "hi/abcdefghijklmnopqrstuvwxyz=/arc-anglerfish-arc2-prod-demo.s3.amazonaws.com/public/JTWX7EUOLJE4FCHYGN2COQAERY.png",
      },
      "promo_items": Object {
        "basic": Object {
          "type": "video",
        },
      },
      "type": "video",
      "website_url": "/food/2021/04/03/best-sourdough-recipes",
    },
  ],
]
`;
